### DISCUSSION TITLE
OTBR acting at BR for Apple Home/Homekit Network

### DISCUSSION THREAD

**Discussion by jodynichols** [openthread/openthread#9003] [Q&A]
Hi team,
Just having a bit of fun with a RPi 3 I had laying around and a nRF52840 dongle I just purchased and have now spun up OTBR on it and it "seemingly" works as expected. I can log in via WebUI and provision/form a new Thread network, but that's where my success ends. My expectation was that I might be able to then use this setup as a BR for my Homekit/Apple Home network and get some of my thread-enabled devices onto a more robust network versus Bluetooth. However, my outcome is that I cannot see this BR in Eve and it doesn't seem to be connected to my Homekit architecture. Is what I am looking to do actually possible? If yes, what am I missing?
Thanks!

**Comment by jwhui** [comment]
Commissioning a Thread device involves transferring Thread credentials to the new device. Apple currently requires those Thread Credentials to be stored within iOS using the [iOS ThreadNetwork APIs](https://developer.apple.com/documentation/threadnetwork/).

**Reply by jodynichols** [reply]
Gotcha. So without writing an app to wrap the api and commission the router, this isn't possible?

**Comment by lancersky** [comment]
Hi @jwhui ,
How do the [iOS ThreadNetwork APIs](https://developer.apple.com/documentation/threadnetwork/) get Thread credentials from a standard OTBR? Is the underlying call via RESTful API like [[rest] add Operational Dataset API](https://github.com/openthread/ot-br-posix/pull/1682)?

**Reply by wgtdkp** [reply]
The iOS API itself doesn't retrieve Thread credentials from a OTBR, you need to write an app which calls maybe the RESTful API to retrieve the credentials from OTBR and then call the iOS API to add it to the system. And then your credentials can be shared to other apps/vendors via the iOS API.

**Reply by wgtdkp** [reply]
The RESTful API can be a candidate, but worth noting that it's plain HTTP without SSL protection, so you should never use it for production

**Reply by jwhui** [reply]
Also worth noting that the existing implementation uses components that are no longer maintained: https://github.com/nodejs/http-parser/issues/522

It would be nice to update ot-br-posix. Contributions are welcome!

**Comment by lboue** [comment]
The entitlement developers need to use the ThreadNetwork API: **com.apple.developer.networking.manage-thread-network-credentials**


**Your app must have this entitlement to invoke the ThreadNetwork API.**  To get the entitlement, request permission from the [ThreadNetwork Framework Entitlement Request](https://developer.apple.com/contact/request/threadnetwork) page.

**Comment by wgtdkp** [comment]
@jwhui Maybe we can open source an Android (and iOS) sample apps to show how the developers can integrate their OTBR-based products with mobile. It will help us make OTBR production-ready.

**Reply by jwhui** [reply]
Reference apps sound good to me. Contributions are welcome!
